init.php

```diff
@@ -193,18 +202,38 @@ public function get_avatar( $avatar = '', $id_or_email, $size = 96, $default = '
 			// Save updated avatar sizes
 			update_user_meta( $user_id, 'basic_user_avatar', $local_avatars );
 
-		} elseif ( substr( $local_avatars[$size], 0, 4 ) != 'http' ) {
-			$local_avatars[$size] = home_url( $local_avatars[$size] );
+		} elseif ( substr( $local_avatars[ $size ], 0, 4 ) != 'http' ) {
+			$local_avatars[ $size ] = home_url( $local_avatars[ $size ] );
 		}
 
 		if ( is_ssl() ) {
 			$local_avatars[ $size ] = str_replace( 'http:', 'https:', $local_avatars[ $size ] );
 		}
 
-		$author_class = is_author( $user_id ) ? ' current-author' : '' ;
-		$avatar       = "<img alt='" . esc_attr( $alt ) . "' src='" . $local_avatars[$size] . "' class='avatar avatar-{$size}{$author_class} photo' height='{$size}' width='{$size}' />";
+		$user_avatar_url = $local_avatars[ $size ];
 
-		return apply_filters( 'basic_user_avatar', $avatar, $user_id );
+		if ( $user_avatar_url ) {
+			$return_args['url'] = $user_avatar_url;
+			$return_args['found_avatar'] = true;
+		}
+
+		return apply_filters( 'basic_user_avatar_data', $return_args );

```

Comment:
Missing phpdoc for hook, something like:

```suggestion
		/**
		 * Allow filtering the avatar data that we are overriding.
		 *
		 * @since 1.0.6
		 *
		 * @param array $return_args The list of user avatar data arguments.
		 */
		return apply_filters( 'basic_user_avatar_data', $return_args );
```